### DISCUSSION TITLE
Use 2.x tools with 1.x framework

### DISCUSSION THREAD

**Discussion by RytoEX** [sparkle-project/Sparkle#2248] [Q&A] [answered]
Is it possible to use the tools (e.g., bin/generate_appcast, bin/generate_keys) from the 2.x versions of Sparkle for applications that still use the 1.x Sparkle framework? Or should you use the tools from the same version the application uses? [This comment](https://github.com/sparkle-project/Sparkle/discussions/2160#discussioncomment-2977249) seems to indicate that the `sign_update` tool from 2.x is safe to use with the 1.x framework, but I wanted to verify that this is the case for all tools.

**Comment by zorgiepoo** [comment, marked as answer]
Yes I would recommend it as more recent versions of the tools are more up to date.

The tools generally have to support generating appcasts (and older delta files) that are readable from older versions of Sparkle. Note though some of the advanced options require users from old versions of your app to be on a minimum Sparkle version and such options are documented as such in the --help page. Also, the way signing updates works hasn't changed (except from a tooling point of view).
